core/thread/id.go

```diff
@@ -301,11 +301,12 @@ func (s IDSlice) Len() int           { return len(s) }
 func (s IDSlice) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
 func (s IDSlice) Less(i, j int) bool { return s[i].str < s[j].str }
 
-// Info holds thread logs and keys.
+// Info holds thread logs, keys and addresses.
 type Info struct {
-	ID   ID
-	Key  Key
-	Logs []LogInfo
+	ID    ID
+	Key   Key
+	Logs  []LogInfo
+	Addrs []ma.Multiaddr
```

Comment:
Added `Addrs` here to represent the thread addresses. It is the `Net` that is responsible for setting this value when appropriate.